silkrpc/core/blocks_test.cpp

```diff
@@ -44,38 +45,44 @@ class MockDatabaseReader : public core::rawdb::DatabaseReader {
 };
 
 TEST_CASE("get_block_number", "[silkrpc][core][blocks]") {
+    SILKRPC_LOG_STREAMS(null_stream(), null_stream());
     const silkworm::ByteView kExecutionStage{stages::kExecution};
     MockDatabaseReader db_reader;
     asio::thread_pool pool{1};
 
     SECTION("kEarliestBlockId") {
-        auto result = asio::co_spawn(pool, get_block_number(kEarliestBlockId, db_reader), asio::use_future);
+        static const std::string EARLIEST_BLOCK_ID = kEarliestBlockId;
```

Comment:
how does this change anything, and why must it be static?  it is an unit test, wouldn't this keep the variable in memory indefinetely?

Reply:
First of all, `get_block_number` is a coroutine taking a `const std::string&` as first formal argument. The calling code in this `SECTION` spawns the coroutine call using `asio::co_spawn` and passing `kEarliestBlockId` (declared `const char*`) as first actual argument. Some automatic conversions and temporary object creations happen here, namely:

1) a temporary `std::string` is created and initialized with a _copy_ of `kEarliestBlockId` content
2) a temporary `std::string_view` representing the entire content of such temporary string is created via conversion operator
3) such temporary string view is passed by constant reference to `get_block_number`

The call to `get_block_number` coroutine will be activated in the `pool` separate thread surely _after_ the end of the `asio::co_spawn` call, which is however the relevant function call to determine the lifetime of the string temporary object created at step 1: hence, such temporary string can be destroyed after the `asio::co_spawn` call and the view can be pointing to nowhere when the `get_block_number` coroutine gets activated. BTW this behavior is undefined and compiler are allowed to perform different kinds of optimizations, in fact the undefined behavior popped just in Clang release mode, GCC was fine

As per `static` specifier, you're right: it's not necessary, I will push a commit to remove it